dbt/adapters/teradata/connections.py

```diff
@@ -337,14 +337,17 @@ def exception_handler(self, sql):
 
     @classmethod
     def get_response(cls, cursor):
-        # There's no real way to get this from teradatasql, so just return "OK"
         num_rows = 0
+        activity = "success"
+        message = "OK"
         if cursor is not None and cursor.rowcount is not None:
-          num_rows = cursor.rowcount
+            num_rows = cursor.rowcount
+            activity = cursor.activityname
+            message = "activity: {}, rows_affected: {}".format(cursor.activityname, cursor.rowcount)
```

Comment:
Varun, did you have discussion on Delete + insert statement with python driver team. Can you please create a table with what activity type is coming for dbt possible scenarios

Reply:
Hi Satish,
It is coming as Delete because Delete+insert is submitted as a single statement in database.
We can make it two different ones, with {% call statement("") %}, if it is required.